test/remote/token_code_tests.js

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const assert = require('insist')
+const config = require('../../config').getProperties()
+const TestServer = require('../test_server')
+const Client = require('../client')()
+const error = require('../../lib/error')
+
+describe('remote tokenCodes', function () {
+  let server, client, email, code
+  const password = 'pssssst'
+  const metricsContext = {
+    flowId: '0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef',
+    flowBeginTime: Date.now()
+  }
+
+  this.timeout(10000)
+
+  before(function() {
+    return TestServer.start(config)
+      .then(s => {
+        server = s
+      })
+  })
+
+  beforeEach(() => {
+    email = server.uniqueEmail()
+    return Client.createAndVerify(config.publicUrl, email, password, server.mailbox)
+      .then(function (x) {
+        client = x
+        assert.ok(client.authAt, 'authAt was set')
+      })
+  })
+
+  it('should error with invalid code', () => {
+    return client.verifyTokenCode('BADCODE', {metricsContext})
+      .then(() => {
+        assert.fail('consumed invalid code')
+      }, (err) => {
+        assert.equal(err.errno, error.ERRNO.INVALID_TOKEN_VERIFICATION_CODE, 'correct errno')
+      })
+  })
+
+  it('should consume valid code', () => {
```

Comment:
For completeness, can you please add a remote test to assert that invalid token codes are rejected, and leave the session unverified?  It'll be good to double-check that end-to-end.